
<img src="http://eagle.fish.washington.edu/cnidarian/skitch/Description_-_Austral_Summer_Institute_XV_1A56F09F.png" alt="Description_-_Austral_Summer_Institute_XV_1A56F09F.png"/>

##Physiological response of marine organisms to environmental change from a genomic perspective

**19-23 January 2015**


_Instructor_:      
Steven Roberts     
robertslab.info     
sr320@uw.edu     
@sr320

_Description_   
Marine environments are continually changing in response to both natural processes and human activities, both having direct physiological impacts on marine organisms. This course will explore the surprising similarities and unique differences in the physiological response of diverse organisms. The main focus will be on functional responses to system stressors; however, the course will also explore how new discoveries in epigenetics is providing insight into phenotypic plasticity and evolutionary implications. A core component of this course will be hands on tutorials on examining high-throughput sequencing data. Specifically, transcriptome (RNA-seq) data from experiments where marine organisms are exposed to environmental change will be analyzed. Bioinformatic instruction will include modules on gene annotation, differential gene expression, and variant detection. There will be an emphasis on open science practices including the use of electronic lab notebooks. 


This is a five day course with AM lectures and hands-on bioinformatics modules in the afternoon. The learning objectives of this course include    
**1) Understanding of fundamental physiological processes    
2) Acquisition of core computing skills    
3) Hands-on experience with genomic datasets**

*Bonus Goal*    
Write two short papers:   
1) Characterization and Annotation of two _Piura_ transcriptomes    
2) Temperature influence on oyster DNA methylation and gene expression

---

#Prerequisites
Before course begins, students should obtain accounts for the following services (if not already signed up for). Students will need access to a computer.

* [GitHub](https://github.com); Clients [Mac](https://mac.github.com) - [Windows](https://windows.github.com)
* [iPlant](http://www.iplantcollaborative.org)
* [SQLShare](https://sqlshare.escience.washington.edu) _Google Account needed_
* [SageMath Cloud](https://cloud.sagemath.com) _can wait for invite_


Please review content below

* [Markdown](https://help.github.com/articles/markdown-basics/)



_Optional_     
Below are recommended software / services that I believe are useful and we will cover in come degree during the course.

* [IPython](http://ipython.org)
* [CoLaboratory Juptyer](http://colaboratory.jupyter.org/welcome/)
* [R](http://www.r-project.org)
* [IGV](http://www.broadinstitute.org/igv/) (registration required)
* [TextWranger (Mac)](http://www.barebones.com/products/textwrangler/)
* [Galaxy](https://usegalaxy.org)

---

#Schedule

##Day 1
Lecture - Introduction (link)

Core Computational Skill - Setup & Bash

Bioinformatics Module - Intro 

##Day 2
Lecture - Basics

CCS - SQL

Bioinformatics Module - Assembly-Annotation

##Day 3
Lecture - Environmental Stress

CCS - R

Bioinformatics Module - RNA-seq

##Day 4
Lecture - Ocean Acidification

CCS - Git

Bioinformatics Module - Genome Browser

##Day 5
Lecture - Epigenetics

CCS - 

Bioinformatics Module - DNA Methylation

##Day 6
Lecture - Disease 

CCS - 

Bioinformatics Module - 

